File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while True:
            temp = n
            dp = 1
            while temp > 0:
                dp *= (temp % 10)
                temp //= 10
            if dp % t == 0:
                return n
            n += 1

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.c
int smallestNumber(int n, int t) {
    
    while(true){
        int temp = n;
        int dp = 1;
        while(temp > 0){
            dp *= (temp % 10);
            temp /= 10;
        }

        if(dp % t == 0){
            return n;
        }
        n++;
    }
}